### PR TITLE
Add ssl endpoint identification algo config

### PR DESCRIPTION
**Pull request by mzamani1**

### CODE REVIEW

**Comment by mzamani1** [comment]
@cricket007 would you be willing to review this?

File: src/main/java/io/dropwizard/kafka/deserializer/StringDeserializerFactory.java [hunk 1]
```diff
@@ -36,15 +36,26 @@ public boolean isEncodingValid() {
...
+        else {
+            deserializerPropertyName = ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG;
+        }
+        final String encodingPropertyName;
+        if (isKey) {
+            encodingPropertyName = "key.deserializer.encoding";
+        }
+        else {
+            encodingPropertyName = "value.deserializer.encoding";
+        }
```
Comment (pstackle) [inline src/main/java/io/dropwizard/kafka/deserializer/StringDeserializerFactory.java:58]: any reason not to combine the `if (isKey)`/`else` blocks?

File: src/main/java/io/dropwizard/kafka/serializer/StringSerializerFactory.java [hunk 2]
```diff
@@ -36,15 +36,26 @@ public boolean isEncodingValid() {
...
+        else {
+            serializerPropertyName = ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG;
+        }
+        final String encodingPropertyName;
+        if (isKey) {
+            encodingPropertyName = "key.serializer.encoding";
+        }
+        else {
+            encodingPropertyName = "value.serializer.encoding";
+        }
```
Comment (pstackle) [inline src/main/java/io/dropwizard/kafka/serializer/StringSerializerFactory.java:58]: same as above. why not combine the if/else check?

File: src/main/java/io/dropwizard/kafka/tracing/TracingFactory.java [hunk 3]
```diff
@@ -50,7 +50,6 @@ public void setRemoteServiceName(final String remoteServiceName) {
         }
 
         return Optional.of(KafkaTracing.newBuilder(tracing)
-                .writeB3SingleFormat(writeB3SingleFormat)
```
Comment (pstackle) [inline src/main/java/io/dropwizard/kafka/tracing/TracingFactory.java:53]: does this make sense to change as part of this PR?

Comment (mzamani1) [reply src/main/java/io/dropwizard/kafka/tracing/TracingFactory.java:53]: It is a no-op now, and is the default behavior as part of a brave update. So this is removing a deprecated method call that does nothing

**Comment by cricket007** [comment]
Sorry! Busy during the day 😅

**Pull request merged**